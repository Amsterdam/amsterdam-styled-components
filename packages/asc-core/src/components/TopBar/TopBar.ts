import styled from '../../styled-components'

type Props = {}

const TopBar = styled.header<Props>`
  display: flex;
  position: relative;
  align-items: center;
  z-index: 0;
  min-height: 54px;
  padding: 0 15px;
`

export default TopBar
