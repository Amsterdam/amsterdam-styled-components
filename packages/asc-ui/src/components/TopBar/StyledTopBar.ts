import styled from 'styled-components'

const defaultBackgroundColor = '#fff'

export type Props = {
  backgroundColor?: string
}

const StyledTopBar = styled.header<Props>`
  display: flex;
  position: relative;
  align-items: center;
  z-index: 0;
  min-height: 54px;
  padding: 0 15px;
  background-color: ${props =>
    props.backgroundColor ? props.backgroundColor : defaultBackgroundColor}
  
  // ie11 fix
  &:after {
    content: '';
    display: block;
    min-height: inherit;
    font-size: 0;
  }
`

export default StyledTopBar


import styled from '../../styled-components'
import { flexboxMinHeightFix } from '../shared/ie-fixes'

type Props = {}

const TopBar = styled.header<Props>`
  display: flex;
  position: relative;
  align-items: center;
  z-index: 0;
  min-height: 54px;
  padding: 0 15px;

  ${flexboxMinHeightFix()}
`

export default TopBar
