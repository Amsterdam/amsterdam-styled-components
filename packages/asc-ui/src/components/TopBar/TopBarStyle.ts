import styled from '@datapunt/asc-core'
import { flexboxMinHeightFix } from '../shared/ie-fixes'
import { breakpoint, color } from '../../utils'
import HeadingStyle from '../Heading/HeadingStyle'

export type Props = {
  backgroundColor?: string
}

const TopBarStyle = styled.header<Props>`
  display: flex;
  position: relative;
  align-items: center;
  z-index: 0;
  padding: 0 10px;
  background-color: ${({ backgroundColor, theme }) =>
    backgroundColor || color('tint', 'level1')({ theme })};
  ${flexboxMinHeightFix()};

  @media screen and ${breakpoint('min-width', 'tabletS')} {
    padding: 0 15px;
  }

  ${HeadingStyle} {
    flex-grow: 1;
    margin-bottom: 0;
  }
`

export default TopBarStyle
