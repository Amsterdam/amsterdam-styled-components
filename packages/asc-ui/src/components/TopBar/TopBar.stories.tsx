import * as React from 'react'
import { storiesOf } from '@storybook/react'
import { ReactComponent as Close } from '@datapunt/asc-assets/lib/Icons/Close.svg'
import TopBar from './TopBar'
import Typography from '../Typography'
import IconButton from '../IconButton'

const SimpleComponent: React.FC<{}> = () => (
  <TopBar>
    <Typography style={{ flexGrow: 1 }} element="h4">
      Lorem ipsum
    </Typography>
    <IconButton aria-label="Close">
      <Close />
    </IconButton>
  </TopBar>
)

storiesOf('TopBar', module).add('default state', () => <SimpleComponent />)
