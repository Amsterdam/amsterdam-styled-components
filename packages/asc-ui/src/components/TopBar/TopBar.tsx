import React, { FunctionComponent } from 'react'
import TopBarStyle, { Props as TopBarStyleProps } from './TopBarStyle'

type Props = TopBarStyleProps

const TopBar: FunctionComponent<Props> = ({ children, ...otherProps }) => (
  <TopBarStyle {...otherProps}>{children}</TopBarStyle>
)

export default TopBar
