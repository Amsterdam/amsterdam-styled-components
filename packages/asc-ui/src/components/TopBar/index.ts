import TopBarStyle from './TopBarStyle'

export { default } from './TopBar'

export const TopBarStyles = { TopBarStyle }
