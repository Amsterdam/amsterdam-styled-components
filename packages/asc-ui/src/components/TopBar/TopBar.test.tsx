import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import 'jest-styled-components'
import TopBar from './TopBar'

jest.useFakeTimers()

describe('TopBar', () => {
  let component: ShallowWrapper

  beforeEach(() => {
    component = shallow(
      <TopBar backgroundColor="red">topbar-content</TopBar>,
    ).dive()
  })

  it('should render', () => {
    expect(component).toMatchSnapshot()
  })
})
