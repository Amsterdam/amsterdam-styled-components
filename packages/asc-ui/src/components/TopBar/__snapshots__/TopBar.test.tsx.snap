// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar should render 1`] = `
<StyledComponent
  backgroundColor="red"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  position: relative;
  align-items: center;
  z-index: 0;
  min-height: 54px;
  padding: 0 15px;
  background-color: ",
          [Function],
          ";
  ",
          "
  &:after {
    content: '';
    display: block;
    min-height: inherit;
    font-size: 0;
  }
",
          ";
",
        ],
      },
      "displayName": "styled.header",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "header",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  topbar-content
</StyledComponent>
`;
